sonic_platform_base/sonic_sfp/sfputilbase.py

```diff
@@ -369,14 +378,59 @@ def read_porttab_mappings(self, porttabfile):
         first = 1
         port_pos_in_file = 0
         parse_fmt_port_config_ini = False
+        parse_fmt_platform_json = False
+
+        parse_fmt_port_config_ini = (os.path.basename(porttabfile) == PORT_CONFIG_INI)
```

Comment:
since portconfig.py already handles port_config.ini and platform.json at the same time, we probably don't need such code here. I'd suggest to use a global function or class method here and reuse the function in both sfputilbase.py and sfputilhelper.py:
```
def read_porttab_mappings(porttabfile):
        logical_to_bcm = {}
        logical_to_physical = {}
        physical_to_logical = {}
        ports, port_alias_map, port_alias_asic_map = parse_port_config_file(porttabfile)
        logical = [None] * len(ports)
        for name, data in ports.items():
            #alias = data['alias']
            index = int(data['index'])
            logical[index] = name
            logical_to_bcm[name] = 'xe{}'.format(index)
            logical_to_physical[name] = [index]
            if index not in physical_to_logical:
                physical_to_logical[index] = [name]
            else:
                physical_to_logical[index].append(name)

        return logical, logical_to_bcm, logical_to_physical, physical_to_logical
```

Reply:
I agree completely. There is a TODO statement about this on line 437 and I have opened an issue regarding this here: https://github.com/Azure/sonic-platform-common/issues/88